### PR TITLE
Reboot cluster from complete outage tests

### PR DESCRIPTION
**Pull request by thedac**
Testing for cold start scenario for MySQL InnoDB Cluster.

### CODE REVIEW

File: zaza/openstack/charm_tests/mysql/tests.py [hunk 1]
```diff
@@ -440,3 +440,104 @@ def test_110_mysqldump(self):
...
+            self.nova_client.servers.start(uuid)
+
+        for unit in zaza.model.get_units(self.application):
+            zaza.model.block_until_unit_wl_status(
+                unit.entity_id,
+                'unknown',
+                negate_match=True)
+
+        logging.debug("Wait till model is idle ...")
+        # XXX If a hook was executing on a unit when it was powered off
```
Comment (ajkavanagh) [inline zaza/openstack/charm_tests/mysql/tests.py]: 'XXX' is this still needed in the comment

Comment (thedac) [reply zaza/openstack/charm_tests/mysql/tests.py]: Removing

File: zaza/openstack/charm_tests/mysql/tests.py [hunk 2]
```diff
@@ -440,3 +440,104 @@ def test_110_mysqldump(self):
...
+    def test_100_cold_start_bootstrap(self):
+        """Bootstrap a non-leader node.
+
+        After bootstrapping a non-leader node, notify bootstrapped on the
+        leader node.
+        """
+        _machines = list(
+            juju_utils.get_machine_uuids_for_application(self.application))
+        # Stop Nodes
+        _machines.sort()
```
[the remark this answers is not in the document]
Comment (thedac) [reply zaza/openstack/charm_tests/mysql/tests.py]: I just tested this. With list().sort() I get None back. By breaking it into two it works. So, for now, I am keeping it.

[the remark this answers is not in the document]
Comment (ajkavanagh) [reply zaza/openstack/charm_tests/mysql/tests.py]: Yes, `list().sort()` sorts the list in place.  `sorted(list())` returns the sorted list leaving the original list untouched.

e.g.

```python
def sort(self) -> None:
    # sort the self thing
```

vs

```python
def sorted(xs: List[Any]) -> List[Any]:
    ys = some_algorithm_that_sorts(xs)
    return ys
```

However, if you prefer the two steps, that's fine too.

**Pull request merged**